src/routes/+layout.svelte

```diff
@@ -16,12 +16,42 @@
     import { loading, requestedMigration } from './store';
     import { parseIfString } from '$lib/helpers/object';
     import Consent from '$lib/components/consent.svelte';
+    import { sdk } from '$lib/stores/sdk';
 
     if (browser) {
         window.VERCEL_ANALYTICS_ID = import.meta.env.VERCEL_ANALYTICS_ID?.toString() ?? false;
     }
 
     onMount(async () => {
+        // handle sources
+        if (isCloud) {
+            const urlParams = $page.url.searchParams;
+            const ref = urlParams.get('ref');
+            const utmSource = urlParams.get('utm_source');
+            const utmMedium = urlParams.get('utm_medium');
+            const utmCampaign = urlParams.get('utm_campaign');
+            let referrer = document.referrer.length ? document.referrer : null;
+
+            // Skip our own
+            if (referrer?.includes('//appwrite.io')) {
+                referrer = null;
+            }
+
+            if (ref || referrer || utmSource || utmCampaign || utmMedium) {
+                try {
+                    await sdk.forConsole.sources.create(
```

Comment:
we shouldnt await it 👍🏻 otherwise its blocking following logic

Reply:
updated